.github/workflows/release.yml

```diff
@@ -0,0 +1,84 @@
+name: Build and Release
+
+env:
+  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  
+on:  
+  push:
+    tags:
+      - '*'
+
+jobs:
+  build-and-upload:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.14.13' 
+      
+      - if: matrix.os == 'windows-latest'
+        name: Install make in windows
+        uses: crazy-max/ghaction-chocolatey@v1
+        with:
+          args: install make
+      
+      - name: Setup Go environment
+        run: make install
+      
+      - name: Read version.txt
+        id: version
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./version.txt
+      
+      # `make build` reads version from version.txt but in Windows it doesn't work therefore passing version explicitly
+      - name: Build go-spacemesh
+        run: make build VERSION=${{ steps.version.outputs.content }} BIN_DIR_WIN=./build
+
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
```

Comment:
Secrets need to be set as an input or an env var to be accessible in a workflow. Have you tested this? I guess the `with` section specifies inputs--are you sure you can access the secrets directly here?

Otherwise you may need to add these three GCP_* secrets to the env section at the top. See https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#accessing-your-secrets for more.

Reply:
@lrettig Yes I have tested the complete pipeline. It successfully builds the binaries and uploads them to our gcloud storage and then creates a draft release. So the secrets are being accessed correctly. Here is the pipeline result: https://github.com/spacemeshos/go-spacemesh/runs/1583958931

Reply:
> LGTM, but please see question I added re: referencing secrets. Also, how are we notified when this runs and cuts a new release? Do you want to add a slack notification here? (See main ci.yml for example.)

We don't need any notifications. We only created tags for releases and it creates a draft release so until someone reviews it and fills change log we won't publish it.